### PR TITLE
Improve validator identifier processing

### LINKED ISSUE #15: Improve error log when index does not exist
# Issue
Wrong error message when validator index does not exist. This leeds to logs not being shown for valid indices aswell.
Release 0.2.0

# Steps to reproduce
Set validator index to some high value like 900000

# Output
13-01-2023 14:53:39 - [ERROR | protocol.request.send_beacon_api_request():] Couldn't read from beacon client. Retry in 5 seconds.

# Expected output
Some message that does not indicate a broken beacon endpoint.

### LINKED ISSUE #19: Add pubkey parsing beside using indices
## Summary
Currently only validator indices can be supplied to the tool. Adding pubkey support improves the tool workflow.

### LINKED ISSUE #26: Add option for validator alias
It would be very helpful to set an alias for a respective index or pub key. This would ease differentiate validators when scanning the logs.

### PR DESCRIPTION
**Pull request by TobiWo**
## Summary
This PR adds/fixes:
* You can now provide indices and/or public keys for validators
  * you can mixup both identifiers in `--validators` 
  * but also in a provided validator file
  * you can now furthermore provide multiple `--validators` flags (e.g. if you want to provide indices and pubkeys but keep it clean in the start command `--validators 123 345 --validators 0x... 0x...`)
* You can decide whether logging messages should use the validator index or pubkey - new cli flag `--log-pubkeys`
* `eth-duties` now filters provided validators for inactive ones at program start
  * fixes a bug where inactive or not present validators would cause `eth-duties` to not read any data
* You can now also provide an alias for one or multiple validators e.g. `--validators 123 "456;validator_1"`
* Documentation was updates
* An example compose was added
* Improve verbosity in case data could not read from beacon client

### Further changes
* Adds the foundation to get rid of the restriction that a user can only provide 300 validators for fetching attestation and sync committee duties. This will be fixed in another PR finally
* Optimizes arguments parsing

## Issues
Closes #15
Closes #19 
Closes #26

### CODE REVIEW

**Review by LuisNaldo7 — CHANGES_REQUESTED** [review]
I am unable to run with multiple indices provided. Providing `--validators "123456, 456789"` will lead to: "Couldn't read from beacon client. Retry in 5 seconds."
Either this is a bug or the documentation is outdated.

Furthermore starting with a single inactive index like `--validators "99999999"` the tool will filter it, print "...99999999 skipped..." and continue logging "Logging next interval... No upcoming duties detected!". A hint that there is no index being monitored would be great.

File: docker/compose.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+# Note: I don't know why but for some reason you cannot provide your validator indices
+# or pubkeys on one line via compose. Therefore you need to add each new index and
+# pubkey via a new line (see below).
+
+version: "3.8"
+name: "Compose for eth-duties helper tool"
+services:
+  eth-duties:
+    container_name: "eth-duties"
+    image: "tobiwo/eth-duties:v0.3.0"
```
Comment (LuisNaldo7) [inline docker/compose.yaml:13]: replace 'image' with

build:
      dockerfile: ./docker/dockerfile
      context: ../

this avoids the need to update the tag every time there is a new release

File: duties/fetcher/parser/validators.py [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+    ]
+    if inactive_validators:
+        __LOGGER.warning(logging.INACTIVE_VALIDATORS_MESSAGE, inactive_validators)
+
+
+def __get_duplicates_with_different_identifiers(
+    provided_valdiators: List[str],
+    complete_validator_identifiers: Dict[str, ValidatorIdentifier],
+) -> List[str]:
+    """Filters for duplicated validators which where provided with different identifiers
```
Comment (LuisNaldo7) [inline duties/fetcher/parser/validators.py:193]: typp: where -> were

**Review by LuisNaldo7 — CHANGES_REQUESTED** [review]

File: docker/compose.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+---
+# This is an example/template compose to add eth-duties to your present docker setup.
+# It assumes that you created a docker network called 'ethereum'.
+# Note: I don't know why but for some reason you cannot provide your validator indices
+# or pubkeys on one line via compose. Therefore you need to add each new index and
+# pubkey via a new line (see below).
+
+version: "3.8"
+name: "Compose for eth-duties helper tool"
```
[the remark this answers is not in the document]
Comment (TobiWo) [reply docker/compose.yaml:9]: Please check if your compose plugin or docker-compose binary is up to date. It works for me.

File: README.md [hunk 4]
```diff
@@ -180,13 +197,13 @@ As always you need to navigate to the root folder of this repository first. Make
 1. Run container
 
     ```bash
-    docker run --rm --name eth-duties tobiwo/eth-duties:latest --validators "123456, 456789" --beacon-node "http://locahost:5052"
+    docker run --rm --name eth-duties tobiwo/eth-duties:latest --validators 123456 456789 --validators 0x98... --beacon-node http://locahost:5052
```
[the remark this answers is not in the document]
Comment (TobiWo) [reply README.md]: I changed several commands as your suggestion is true for all longer commands. Thanks

**Review by LuisNaldo7 — APPROVED** [review]

**Review by LuisNaldo7 — APPROVED** [review]

**Pull request merged**